COMPATIBILITY.md

```diff
@@ -0,0 +1,102 @@
+# `rules_rust` backwards compatibility policy
+
+This document defines the backwards compatibility policy for rules\_rust and
+defines the process for making compatibility breaking changes. Any exception to
+this process will have to be thoroughly discussed and there will have to be a
+consensus between maintainers that the exception should be granted.
+
+## What is a compatibility breaking change?
+
+A backwards compatible change has the property that a build that is green,
+correct, is using stable APIs of `rules_rust,` and is using a supported version
+of Bazel before the change is still green and correct after the change. In
+practice it means that users should not have to modify their project source
+files, their `BUILD` files, their `bzl` files, their `rust_toolchain`
+definitions, their platform definitions, or their build settings.
+
+A backwards compatible release has the property that it only contains backward
+compatible changes.
+
+`rules_rust` follow [SemVer 2.0.0](https://semver.org/). `rules_rust` promise
+that all minor version number and patch version number releases only contain
+backwards compatible changes.
+
+`rules_rust` don't make any guarantees about compatibility between a released
+version and the state of the `rules_rust` at HEAD.
+
+Backwards incompatible changes will have to follow a process in order to be
+released. Users will be given at least one release where the old behavior is
+still present but deprecated to allow smooth migration of their project to the
+new behavior.
+
+## What Bazel versions are supported?
+
+`rules_rust` support the current Bazel LTS at the time of a `rules_rust`
+release.
+
+Support for the current Bazel rolling release is on the best effort basis. If
+the CI build of `rules_rust` against the current Bazel rolling release is green,
+it has to stay green. If the build is already red (because the new Bazel rolling
```

Comment:
> If the build is already red (because the new Bazel rolling release had incompatible changes that broke `rules_rust`), it is acceptable to merge PRs leaving the build red as long as the reason for failure remains the same

Just out of curiosity, is there a way to manually trigger a build so Bazel incompatibility can be independently verified by a user?

Reply:
Similarly, is there a way we can trigger CI when new rolling releases come out to notify us of upcoming failure?

Reply:
Buildkite builds can only be triggered by users who are explicitly allowed to (buildkite has "teams", and scentini@ and I are the Bazel's "rust" team there, so we can trigger builds manually). For random users this will not be allowed, but maintainers of rules_rust quite likely yes.

> Similarly, is there a way we can trigger CI when new rolling releases come out to notify us of upcoming failure?

Yes, I hope to have such pipeline.

Reply:
(resolving now, feel free to reopen)